src/sage/modular/quasimodform/ring.py

```diff
@@ -778,3 +778,47 @@ def from_polynomial(self, polynomial):
             raise ValueError("the number of variables (%s) of the given polynomial cannot exceed the number of generators (%s) of the quasimodular forms ring" % (nb_var, self.ngens()))
         gens_dict = {poly_parent.gen(i):self.gen(i) for i in range(0, nb_var)}
         return self(polynomial.subs(gens_dict))
+
+    def basis_of_weight(self, weight):
+        r"""
+        Return a basis of elements generating the subspace of the given
+        weight.
+
+        INPUT:
+
+        - ``weight`` (integer) -- the weight of the subspace
```

Comment:
```suggestion
        - ``weight`` -- an integer, the weight of the subspace
```
The type goes after the double dash.

Reply:
I don't think this is an official rule, in fact a quick look at the documentation shows many examples of this type of format.